zetaclient/chains/bitcoin/observer/observer.go

```diff
@@ -634,6 +634,26 @@ func GetTxResultByHash(
 	return hash, txResult, nil
 }
 
+// GetBlockHeightByTxHash gets the block height by block hash
+func GetBlockHeightByHash(
+	rpcClient interfaces.BTCRPCClient,
+	hash string,
+) (int64, error) {
+	// decode the block hash
+	var blockHash chainhash.Hash
+	err := chainhash.Decode(&blockHash, hash)
+	if err != nil {
+		return 0, errors.Wrapf(err, "GetBlockHeightByHash: error decoding block hash %s", hash)
+	}
+
+	// get block by hash
+	block, err := rpcClient.GetBlockVerbose(&blockHash)
```

Comment:
That was what I wanted to use. Unfortunately, the bitcoin block header doesn't contain height. The performance is now `180 ms` per RPC call for a typical bitcoin block. This could be improved by employing a cache in future: https://github.com/zeta-chain/node/issues/2252